### ISSUE TITLE
Accessing the store from outside react

### ISSUE THREAD

**Issue by mitsuhiko** [bcherny/undux#51]
How can I get the underlying store from `createConnectedStore` now that `createStore` is deprecated? More importantly how do I use this without react now?

**Comment by bcherny** [comment]
Hey @mitsuhiko! You can still use `createStore` for now - if there's interest in keeping it around, we may just rename it to `createSingletonStore` in the future.

For React users, the big benefit of `createConnectedStore` is it means stores can now get garbage collected when the view unmounts. The way it works is it doesn't actually create a store (a bit of a misnomer), but it gives you the two pieces you need to let React construct a store instance for you. For non-React users, I'd love to see similar bindings for your framework of choice.

Which framework are you using? Or, just raw JS?

**Comment by mitsuhiko** [comment]
@bcherny I am actually using it with react now but I have some utility code that is itself not using react. For instance my API layer accesses auth tokens from the store and wants to notify the store if the requests fail because the user session was invalidated. That API layer is largely disconnected from my react code.

**Comment by giddens9527** [comment]
I'm also facing similiar problem. I was trying to build extensions/plugins for my react app and decided to expose the undux store for non-react plugins/extensions to mutate. It works great now but I'm not sure  about the performance when there are multiple plugins trying to update the same store.

**Comment by bcherny** [comment]
@giddens9527 Can you talk a little more about your use case, and what your specific performance concern is? A code sample would be helpful.

**Comment by matheusgrieger** [comment]
I have faced the same issue. I created some services to share commonly used code across many components, but in one case I need some values that are present inside the store. The only way without using `createStore` is by passing the store prop as an argument to the service function, which is cumbersome. Can' `createConnectedStore` also output the store itself, the same way as the prop is passed to components? If not, I am fairly certain that the `createStore`, as only solution, would be valuable to keep around and un-deprecate it.

**Comment by matheusgrieger** [comment]
By the way, when using `createStore` and `connect`, how do I get the `Container` used to wrap the application?

**Comment by bcherny** [comment]
> Can' `createConnectedStore` also output the store itself, the same way as the prop is passed to components?

No. `createConnectedStore` is different than `createStore`, because it lets React manage and instantiate (via Context) the store for you. This has two big benefits:

1. It avoids top-level references to store instances, meaning they can get properly garbage collected when they're no longer used.
2. It lets you create multiple instances of a store.

> By the way, when using `createStore` and `connect`, how do I get the `Container` used to wrap the application?

`createStore` doesn't need a Container, since it creates a singleton store.

**Comment by matheusgrieger** [comment]
So I can simply use the `withStore` HOC on my components?
